src/borg/repository.py

```diff
@@ -396,7 +396,8 @@ def open(self, path, exclusive, lock_wait=None, lock=True):
         try:
             st = os.stat(path)
         except FileNotFoundError:
-            raise self.DoesNotExist(path)
+            self.close()
+            raise self.InvalidRepository(self.path)
```

Comment:
overall, you changed the "does not exist" (which is true) into a "invalid" (which is questionable) exception.

what's the point?

Reply:
also, please do not merge master into your branch if there is no reason to do so.